specs/simpleserialize.md

```diff
@@ -0,0 +1,302 @@
+return byte_length + value
+```
+
+#### List/Vectors
+
+1. Get the number of raw bytes to serialize: it is `len(list) * sizeof(element)`.
+   * Encode that as a `4-byte` **big endian** `uint32`.
+2. Append your elements in a packed manner.
+
+* *Note on efficiency*: consider using a container that does not need to iterate over all elements to get its length. For example Python lists, C++ vectors or Rust Vec.
+
+**Example in Python**
+
+```python
+serialized_list_string = ''
+
+for item in value:
+   serialized_list_string += serialize(item)
+
+serialized_len = (len(serialized_list_string).to_bytes(LENGTH_BYTES, 'big'))
+
+return serialized_len + serialized_list_string
+```
+
+### Deserialize/Decode
+
+The decoding requires knowledge of the type of the item to be decoded. When
+performing decoding on an entire serialized string, it also requires knowledge
+of what order the objects have been serialized in.
+
+Note: Each return will provide ``deserialized_object, new_index`` keeping track
+of the new index.
+
+At each step, the following checks should be made:
+
+| Check Type               | Check                                                     |
+|:-------------------------|:----------------------------------------------------------|
+| Ensure sufficient length | ``length(rawbytes) > current_index + deserialize_length`` |
+
+#### uint: 8/16/24/32/64/256
+
+Convert directly from bytes into integer utilising the number of bytes the same
+size as the integer length. (e.g. ``uint16 == 2 bytes``)
+
+All integers are interpreted as **big endian**.
+
+```python
+byte_length = int_size / 8
+new_index = current_index + int_size
+return int.from_bytes(rawbytes[current_index:current_index+int_size], 'big'), new_index
+```
+
+#### Address
+
+Return the 20 bytes.
+
+```python
+new_index = current_index + 20
+return rawbytes[current_index:current_index+20], new_index
+```
+
+#### Hash
+
+##### Hash32
+
+Return the 32 bytes.
+
+```python
+new_index = current_index + 32
+return rawbytes[current_index:current_index+32], new_index
+```
+
+##### Hash96
+
+Return the 96 bytes.
+
+```python
+new_index = current_index + 96
+return rawbytes[current_index:current_index+96], new_index
+```
+
+##### Hash97
+
+Return the 97 bytes.
+
+```python
+new_index = current_index + 97
+return rawbytes[current_index:current_index+97], new_index
+```
+
+
+#### Bytes
+
+Get the length of the bytes, return the bytes.
+
+```python
+bytes_length = int.from_bytes(rawbytes[current_index:current_index + LENGTH_BYTES], 'big')
+new_index = current_index + LENGTH_BYTES + bytes_lenth
+return rawbytes[current_index + LENGTH_BYTES:current_index+ LENGTH_BYTES +bytes_length], new_index
+```
+
+#### List/Vectors
+
+Deserialize each object in the list.
+1. Get the length of the serialized list.
+2. Loop through deserializing each item in the list until you reach the
+entire length of the list.
+
+
+| Check type                          | code                                  |
```

Comment:
Use "Check to perform" as in previous tables, and add the assertion in the sample code